### ISSUE TITLE
OpenCL has no member enqueueBarrierWithWaitList()

### ISSUE THREAD

**Issue by GPadley** [HPCE/hpce-2017-cw3#87]
When trying to compile both test_opencl, v4 and v5 I am unable to use enqueueBarrierWithWaitList(); Giving the error below.

``
 error: 'class cl::CommandQueue' has no member named 'enqueueBarrierWithWaitList'; did you mean 'enqueueBarrier'?
``

Changing this to enqueueBarrier() enables the programs to compile and run correctly on my machine. However, when @jjd06 run the tests, which function call will be used within the AWS instance or is it okay to use enqueueBarrier() instead?

**Comment by jjd06** [comment]
No. `enqueueBarrierWithWaitList()` is an OpenCL 1.2 feature, and should be usable (and used). Make sure that you have an up-to-date OpenCL install on the machine you're using.

**Comment by GPadley** [comment]
I've been attempting to do this on my Windows 10 machine and have been unable to do so. I have changed my repo such that v4 and v5 are written with enqueueBarrierWithWaitList() however I am also unable to open up a g3 EC2 instance to test as free tier AWS account is blocking this.

**Comment by jjd06** [comment]
The high-performance instances are not available for free. Have you requested credit through AWS Educate? There is a lead time for this. There's also typically a small delay (a few hours) in getting an instance the first time you try to launch one.

**Comment by GPadley** [comment]
My account was already verified, limits are set on the free tier account on what instances you are able to use. I had to contact AWS to have the limit moved from 0 to 1 to allow me to use the specific instance.

**Comment by jjd06** [comment]
Ah, good to know. So are the two issues (out-of-date OpenCL library and instance access) resolved now?
